ReactNativeClient/lib/models/Note.js

```diff
@@ -85,21 +86,16 @@ class Note extends BaseItem {
 		if (body && body.length) {
 			const lines = body.trim().split('\n');
 			let output = lines[0].trim();
-			const mdLinkRegex = /!?\[(.+)\]\(.+\)/;
-			if (output.match(mdLinkRegex)) {
-				output = output.replace(mdLinkRegex, '$1');
-			} else {
-				// Remove the first #, *, etc.
-				while (output.length) {
-					const c = output[0];
-					if (['#', ' ', '\n', '\t', '*', '`', '-'].indexOf(c) >= 0) {
-						output = output.substr(1);
-					} else {
-						break;
-					}
+			// Remove the first #, *, etc.
+			while (output.length) {
+				const c = output[0];
+				if (['#', ' ', '\n', '\t', '*', '`', '-'].indexOf(c) >= 0) {
+					output = output.substr(1);
+				} else {
+					break;
 				}
 			}
-			return output.substr(0, 80).trim();
+			return mdUtils.filterLink(output).substr(0, 80).trim();
```

Comment:
Why do you put half the logic in your function, and you leave the other half here? Wouldn't it be logical to move everything to the function?

Reply:
Alright. So I'll make one function to filter out markdown from the title and use it here. Just one doubt, would it make sense to remove the `defaultTitleFromBody` function altogether and do the entire thing in `defaultTitle` itself?